index.js

```diff
@@ -138,7 +138,7 @@ class Hyperdiscovery extends EventEmitter {
       self._lock(release => {
         _add(dkey)
           .then(() => release())
-          .catch(err => release(err))
+          .catch(err => release(null, err))
```

Comment:
you risk calling release twice by doing it this case in case release triggers something that throws

Reply:
(oh i guess that's unrelated to your PR)

Reply:
I see what you mean. Should I edit my PR to account for that anyway?